src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -79,6 +84,76 @@ private static String createNewIndex(PineconeControlPlaneClient controlPlaneClie
         return indexName;
     }
 
+    public static IndexModel waitUntilIndexIsReady(PineconeControlPlaneClient controlPlaneClient, String indexName, Integer totalMsToWait) throws InterruptedException {
```

Comment:
I added this because using `isIndexReady` wasn't sufficient for waiting on indexes that are created from collections.

They seemingly can take a long time, sometimes longer than 2 minutes. `isIndexReady` only allows you to control the backoff, which is clunky when trying to specify a more specific amount of time to wait.

We do something similar to this for polling in the other clients.

Reply:
Adding specific amount of time is like hard coding (if there's some platform issue, it'll fail), so can we pass a higher backoff instead?